crates/cheatcodes/src/inspector.rs

```diff
@@ -941,6 +941,19 @@ where {
 
         // Apply our prank
         if let Some(prank) = &self.prank {
+            // Apply delegate call. call.caller will not equal prank.prank_caller
+            if let CallScheme::DelegateCall = call.scheme {
```

Comment:
should we handle `CallScheme::ExtDelegateCall` here too?

Reply:
Good catch, yes.